src/zenml/models/v2/base/filter.py

```diff
@@ -248,31 +246,23 @@ def generate_query_conditions_from_column(self, column: Any) -> Any:
 
         if self.operation == GenericFilterOps.ONEOF:
             assert isinstance(self.value, list)
-            # Convert the list of values to a list of json strings
-            json_list = (
-                [json.dumps(v) for v in self.value]
-                if self.json_encode_value
-                else self.value
-            )
-            return column.in_(json_list)
+            return column.in_(self.value)
 
-        # Don't convert the value to a json string if the operation is contains
-        # because the quotes around strings will mess with the comparison
         if self.operation == GenericFilterOps.CONTAINS:
             return column.like(f"%{self.value}%")
-
-        json_value = (
-            json.dumps(self.value) if self.json_encode_value else self.value
-        )
-
         if self.operation == GenericFilterOps.STARTSWITH:
```

Comment:
Now thinking about this, if the column does not contain json encoded values, this would simply return false results, correct?

If this is simply a string column and I pass in `startswith:a`, it would give me results I do not expect. Maybe it is worth having this as a special handling only for certain columns, not the generic string filter.

Reply:
What do you mean by false results? 

A great example that I can think of, in terms of strings columns that do not contain json encoded values, is the stack name. If I do `zenml stack list --name=startswith:def`, it still returns the `default` stack. The potential problem is, if we ever create a stack with the name `"default"`, that would get return as well. Is that what you mean?

Reply:
Yep exactly

Reply:
The only column that I can think of (where we have this type and value in separate columns, and every value is JSON encoded) is the value column in the RunMetadataSchema. Can you think of anything else?

Also, if I separate this case (and only add the `or_` conditions if we are dealing with that column), this would fix it for most filters. However, we will still run into the same issue with the run metadata filtering. At this moment in the code, all I am getting from the user is a string. If I ran two pipelines with `log_metadata("random_key": 1)` and `log_metadata("random_key": "1")` and people filter by `run_metadata=["random_key:startswith:1"]`, they will get both. Is that fine for you?

Reply:
No can't think of any other column, which makes this even more "important": We're essentially implementing the `StrFilter` (which is used for 100s of columns) to work correctly with 1 column instead of the 99 others.

I think the second case with `"1"` and `1` in this column, there isn't any quick solution to it so I think it should be fine

Reply:
I have addressed these issues in my latest commit.